lib/mongo_ecto/connection.ex

```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
       fun.()
     rescue
       e in Mongo.Error ->
-        stacktrace = System.stacktrace
         case e do
           %Mongo.Error{code: 11000, message: msg} ->
             {:invalid, constraint(msg)}
           other ->
+            stacktrace = System.stacktrace
```

Comment:
Do we really need to acquire stacktrace when we're using it only in this node? It increases readability or maybe it's pattern that should we use?